File: src/main/java/org/apache/fineract/client/services/SelfShareProductsApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import io.reactivex.Observable;
import io.reactivex.Completable;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;


import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface SelfShareProductsApi {
  /**
   * 
   * 
   * @param clientId  (optional)
   * @param offset  (optional)
   * @param limit  (optional)
   * @return Observable&lt;String&gt;
   */
  @GET("self/products/share")
  Observable<String> retrieveAllProducts1(
    @retrofit2.http.Query("clientId") Long clientId, @retrofit2.http.Query("offset") Integer offset, @retrofit2.http.Query("limit") Integer limit
  );

  /**
   * 
   * 
   * @param productId  (required)
   * @param type  (required)
   * @param clientId  (optional)
   * @return Observable&lt;String&gt;
   */
  @GET("self/products/share/{productId}")
  Observable<String> retrieveProduct1(
    @retrofit2.http.Path("productId") Long productId, @retrofit2.http.Path("type") String type, @retrofit2.http.Query("clientId") Long clientId
  );

}

File: src/main/kotlin/org/openapitools/client/apis/SelfShareProductsApi.kt
package org.openapitools.client.apis

import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query


interface SelfShareProductsApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param clientId  (optional)
     * @param offset  (optional)
     * @param limit  (optional)
     * @return [kotlin.String]
     */
    @GET("v1/self/products/share")
    suspend fun retrieveAllProducts1(
        @Query("clientId") clientId: Long? = null,
        @Query("offset") offset: Int? = null,
        @Query("limit") limit: Int? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param productId
     * @param type
     * @param clientId  (optional)
     * @return [kotlin.String]
     */
    @GET("v1/self/products/share/{productId}")
    suspend fun retrieveProduct1(
        @Path("productId") productId: Long,
        @Path("type") type: String,
        @Query("clientId") clientId: Long? = null
    ): String

}
